src/Services/IncomeService.cs

```diff
@@ -0,0 +1,109 @@
+using Domain.DTOs;
+using Domain.Entities;
+
+using ExpenseTrackerGroup3.Repositories.Interfaces;
+using ExpenseTrackerGroup3.Services.Interfaces;
+
+namespace ExpenseTrackerGroup3.Services;
+
+public class IncomeService : IIncomeService
+{
+    private readonly IIncomeRepository _incomeRepository;
+    private readonly IUserRepository _userRepository;
+
+    public IncomeService(IIncomeRepository incomeRepository, IUserRepository userRepository)
+    {
+        _incomeRepository = incomeRepository;
+        _userRepository = userRepository;
+    }
+
+    public async Task<Income> AddIncomeAsync(Guid userId, CreateIncome income)
+    {
+        var user = await _userRepository.GetByIdAsync(userId);
+
+        if (user == null)
+        {
+            throw new ArgumentException("User not found");
+        }
+
+        var newIncome = new Income
+        {
+            Id = Guid.NewGuid(),
+            UserId = userId,
+            Amount = income.Amount,
+            Source = income.Source,
+            CreatedAt = DateTime.Now
+        };
+
+        var success = await _incomeRepository.CreateAsync(newIncome);
+
+        if (!success)
+        {
+            throw new Exception("Failed to create income");
+        }
+
+        return newIncome;
+    }
+
+    public async Task<IEnumerable<Income>> GetIncomesByUserIdAsync(Guid userId)
+    {
+
+        User? user = await _userRepository.GetByIdAsync(userId);
+
+        if (user == null) 
+        {
+            throw new ArgumentException("User not found");
+        }
+
+        IEnumerable<Income> income = await _incomeRepository.GetAllAsync();
+
+        return income.Where(i => i.UserId == userId);
+    }
+
+    public async Task<IEnumerable<Income>> GetMonthlyIncomeByUserId(Guid userId, DateTime month)
+    {
+        User? user = await _userRepository.GetByIdAsync(userId);
+
+        if (user == null)
+        {
+            throw new ArgumentException("User not found");
+        }
+
+       IEnumerable<Income> incomes = await _incomeRepository.GetAllAsync();
+
+       return incomes.Where(i => i.UserId == userId && i.CreatedAt.Month == month.Month);
+    }
+
+    public async Task<Income> UpdateIncomeAsync(Guid userId, UpdateIncome income)
```

Comment:
Inside the body of the endpoint there should not be the option to change the id since it is the same and neither the creation date:

![image](https://github.com/user-attachments/assets/20dbe78c-8f94-4fe9-9c82-f2fb298d72a6)